test/dummies/BadContractInteractor.ts

```diff
@@ -1,18 +1,16 @@
 import { TransactionReceipt } from 'web3-core'
 
-import ContractInteractor, { Web3Provider } from '../../src/relayclient/ContractInteractor'
 import RelayRequest from '../../src/common/EIP712/RelayRequest'
-import { GSNConfig } from '../../src/relayclient/GSNConfigurator'
-import { LoggerInterface } from '../../src/common/LoggerInterface'
+import ContractInteractor, { ConstructorParams } from '../../src/common/ContractInteractor'
 
 export default class BadContractInteractor extends ContractInteractor {
   static readonly message = 'This is not the contract you are looking for'
   static readonly wrongNonceMessage = 'the tx doesn\'t have the correct nonce'
 
   private readonly failValidateARC: boolean
 
-  constructor (provider: Web3Provider, logger: LoggerInterface, config: GSNConfig, failValidateARC: boolean) {
-    super(provider, logger, config)
+  constructor (constructorParams: ConstructorParams, failValidateARC: boolean) {
```

Comment:
nitpic: if you define named params, then why do you add new unnamed one?
(I know, its only a test. i think you can extend type in-place to add new member. I just don't know how.)

Reply:
This file has to go and be replaced by sinon mock objects, I just had no time to get to it